citro3d/src/buffer.rs

```diff
@@ -46,6 +49,62 @@ impl Slice<'_> {
     pub fn info(&self) -> &Info {
         self.buf_info
     }
+
+    /// Get an index buffer for this slice using the given indices.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - any of the given indices are out of bounds.
+    /// - the given slice is too long for its length to fit in a `libc::c_int`.
+    pub fn index_buffer<I>(&self, indices: &[I]) -> Result<Vec<I, LinearAllocator>, Error>
```

Comment:
Thinking about it more this might be better off returning an `IndexSlice` or something that's tied to the lifetime of `self.buf_info`, to ensure the index buffer doesn't outlive the vertex buffer. I might give that a shot if I can figure out a way to do it that isn't too awkward to use